src/System.Private.CoreLib/shared/System/Number.Formatting.cs

```diff
@@ -1852,7 +1853,7 @@ internal static unsafe void NumberToStringFormat(ref ValueStringBuilder sb, ref
                 {
                     number.Scale += scaleAdjust;
                     int pos = scientific ? digitCount : number.Scale + digitCount - decimalPos;
-                    RoundNumber(ref number, pos);
+                    RoundNumber(ref number, pos, isCorrectlyRounded: false);
```

Comment:
Custom-numeric format strings come through here and the buffer will generally have more digits than necessary. In which case we are reliant on `RoundNumber` to both trim trailing zeros and correctly rounding it to the number of digits specified by the custom-numeric format string.